test/server/graphql/v2/mutation/AccountMutations.test.ts

```diff
@@ -446,4 +449,138 @@ describe('server/graphql/v2/mutation/AccountMutations', () => {
       expect(collective.data.policies).to.be.empty;
     });
   });
+
+  describe('sendMessage', () => {
+    const sendMessageMutation = gqlV2/* GraphQL */ `
+      mutation SendMessage($account: AccountReferenceInput!, $message: NonEmptyString!, $subject: String) {
+        sendMessage(account: $account, message: $message, subject: $subject) {
+          success
+        }
+      }
+    `;
+
+    const message = 'Hello collective, I am reaching out to you for testing purposes.';
+
+    let sandbox, sendEmailSpy, collectiveWithContact, collectiveWithoutContact;
+
+    before(async () => {
+      sandbox = createSandbox();
+      collectiveWithContact = await fakeCollective({
+        name: 'Test Collective',
+        admin: adminUser,
+        settings: { features: { contactForm: true } },
+      });
+      collectiveWithoutContact = await fakeCollective({
+        admin: adminUser,
+        settings: { features: { contactForm: false } },
+      });
+
+      sendEmailSpy = sandbox.spy(emailLib, 'sendMessage');
+    });
+
+    after(() => sandbox.restore());
+
+    afterEach(() => {
+      sendEmailSpy.resetHistory();
+    });
+
+    it('sends the message by email', async () => {
+      const result = await graphqlQueryV2(
+        sendMessageMutation,
+        {
+          account: { id: idEncode(collectiveWithContact.id, 'account') },
+          message,
+          subject: 'Testing',
+        },
+        randomUser,
+      );
+
+      expect(result.errors).to.not.exist;
+      expect(result.data.sendMessage.success).to.equal(true);
+
+      await waitForCondition(() => sendEmailSpy.callCount === 1);
+      expect(sendEmailSpy.callCount).to.equal(1);
+      expect(sendEmailSpy.args[0][1]).to.equal(
+        `New message from ${randomUser.collective.name} on Open Collective: Testing`,
+      );
+      expect(sendEmailSpy.args[0][2]).to.include(message);
+    });
+
+    it('cannot inject code in the email (XSS)', async () => {
+      const code = '<script>console.log("XSS")</script>';
+      const xssUser = await fakeUser({ name: 'XSS User' }, { name: 'XSS Collective', slug: 'xss-collective' });
+      const result = await graphqlQueryV2(
+        sendMessageMutation,
+        {
+          account: { id: idEncode(collectiveWithContact.id, 'account') },
+          message: message + code,
+          subject: 'Testing',
+        },
+        xssUser,
+      );
+
+      expect(result.errors).to.not.exist;
+      expect(result.data.sendMessage.success).to.equal(true);
+
+      await waitForCondition(() => sendEmailSpy.callCount === 1);
+      expect(sendEmailSpy.callCount).to.equal(1);
+      expect(sendEmailSpy.args[0][1]).to.equal(`New message from XSS Collective on Open Collective: Testing`);
+      expect(sendEmailSpy.args[0][2]).to.include(message);
+      expect(sendEmailSpy.args[0][2]).to.not.include('<script>');
```

Comment:
If, somehow, the attack turns out creating a code like:

```
<code attr="1">test</code>
```

This test won't detect it. Can we use the full text here, i.e.:

```suggestion
      expect(sendEmailSpy.args[0][2]).to.eq(fullText);
```

Reply:
@Betree here full text is actually the whole HTML, that's why I didn't add that check.

We may also want to improve existing XSS test cases then:

https://github.com/opencollective/opencollective-api/blob/beed05187b621d24526107362377414716c71429/test/server/lib/notifications.test.ts#L185

<details>
<summary>  sendEmailSpy.args[0][2]   </summary>

```html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
       <html xmlns="http://www.w3.org/1999/xhtml">
       <head>
         <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
         <title></title>
             <td></td>
             <td>
       
       <p style="color: #494B4D; font-family: 'Helvetica Neue'; line-height: 18px; font-size: 17px; padding: 1em
       Hi Test Collective 127ae549,
         <br><br>
       <a href="http://localhost:3000/user-24640536" style="text-decoration: none; color: #297EFF;">User Namec778c9a9</a> just sent you a message on Open Collective. Simply reply to this email to reply to the sender.
       </p>
       
       <table style="width: 100%;">
       </table><!-- 100% width table -->
       <!-- OpenCollective.com -->
       </body>
       
      </html>
```
</details>

Reply:
@Betree Done ✅